contents/dart/Dart.md

```diff
@@ -0,0 +1,228 @@
+```dart
+/// Sets the [bold] and [hidden] flags ...
+void enableFlags({bool bold, bool hidden}) {...}
+```
+
+Note that if named parameters are used, the order no longer matters, you can write `enableFlags(hidden: false, bold: true);` as well. This feature is especially useful for declarative UI as the properties of a widget are, most of the time, parameters passed to the constructor of the object. Below is a concrete example in Flutter. Imagine how messy and frustrating the code would look like if the parameters are not named and we have to refer to the documentation for the exact order of them!
+
+```dart
+// Copyright 2018 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+void main() => runApp(MyApp());
+
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      title: 'Welcome to Flutter',
+      home: Scaffold(
+        appBar: AppBar(
+          title: Text('Welcome to Flutter'),
+        ),
+        body: Center(
+          child: Text('Hello World'),
+        ),
+      ),
+    );
+  }
+}
+```
+
+<panel header="DartPad example"><iframe style="width: 100%; height: 400px;" src="https://dartpad.dev/embed-flutter.html?id=719143b1a625dbdefb853c96de0ddbc2&split=50"></iframe></panel>
+
+Note that **named** parameters are a type of [**optional** parameters](https://dart.dev/guides/language/language-tour#optional-parameters) (i.e. you can omit those parameters and they will be assigned `null` or the default value upon entry of the function). You can also make use of [`@required`](https://pub.dev/documentation/meta/latest/meta/required-constant.html) annotation to make certain named parameters compulsory.
+
+### Benefit (advanced): Interoperability
+You can call C and JavaScript code from Dart using [dart:ffi](https://api.dart.dev/dev/2.8.0-dev.7.0/dart-ffi/dart-ffi-library.html) for C or [js](https://pub.dev/packages/js) for JavaScript. These libraries translate data types from other languages to native Dart data types for mixed operation.
+
+## How to Get Started with Dart?
+
+Dart is well-documented and you should find solutions to most of your Dart problems on [their website](https://dart.dev/guides). 
+
+* Installation: You may get the Dart SDK [here](https://dart.dev/get-dart).
+* Syntax tour: [A tour of the Dart language](https://dart.dev/guides/language/language-tour)
+* Essential concepts in Dart:
+  * [Asynchronous programming using `futures`, `async`, `await`](https://dart.dev/codelabs/async-await)
+  * [Asynchronous programming using streams](https://dart.dev/tutorials/language/streams)
+  * [Dart type system](https://dart.dev/guides/language/sound-dart)
+
+Do note that a lot of the knowledge learnt from programming with Dart is applicable to other languages as well (e.g. asynchronous programming, generics, etc.).
```

Comment:
I think this is true for any language, could remove this since it dosen't tie to "how to get started with dart", but is more of a benefit